src/connection.h

```diff
@@ -255,8 +256,18 @@ static inline int connSetWriteHandlerWithBarrier(connection *conn, ConnectionCal
     return conn->type->set_write_handler(conn, func, barrier);
 }
 
-static inline void connShutdown(connection *conn) {
-    conn->type->shutdown(conn);
+static inline void connShutdown(connection *conn, int is_rdb_channel) {
```

Comment:
This API seems weird to me, should `is_rdb_channel` be a flag in the connection and default to `0`. With that the `connShutdown` API doesn't need to change and deal with the conn->flag and take the decision?

Reply:
`connShutdown` API does seems a bit unconventional. However, introducing a new flag in the connection struct solely for this purpose might not be the best approach. By keeping the logic encapsulated within `connShutdown`, we can avoid polluting the connection struct with additional flags, making the code easier to reason about and debug. 
IMO local variables and self-contained functions are generally better in terms of code organization and maintainability.

Reply:
Instead of is_rdb_channel, I think we should have a flag like, "no_block" or "force", which indicates the intent of the operation. I think the best way to think about this is what if someone is building a new type of connection, how will they think about what to do about this connection.